src/main/java/sirius/db/mixing/query/QueryCompiler.java

```diff
@@ -13,27 +13,76 @@
 import sirius.db.mixing.Mapping;
 import sirius.db.mixing.Property;
 import sirius.db.mixing.properties.BaseEntityRefProperty;
+import sirius.db.mixing.properties.LocalDateProperty;
+import sirius.db.mixing.properties.LocalDateTimeProperty;
+import sirius.db.mixing.properties.LocalTimeProperty;
 import sirius.db.mixing.query.constraints.Constraint;
 import sirius.db.mixing.query.constraints.FilterFactory;
 import sirius.kernel.commons.Strings;
+import sirius.kernel.commons.Tuple;
 import sirius.kernel.commons.Value;
 import sirius.kernel.di.GlobalContext;
 import sirius.kernel.di.std.Part;
 
 import javax.annotation.Nullable;
 import java.io.StringReader;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalUnit;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
- * Parses an SQL like query and compiles it into a {@link Constraint}.
+ * Parses a query and compiles it into a {@link Constraint}.
```

Comment:
This commit is really hard to review as it includes refactorings and logic changes in one. Also it breaks the build. Could this be revised?